### PR TITLE
feat(code): default code to snake case of name

### PR DESCRIPTION
**Pull request by ajpawlicki**
## What

Defaults `code` to decamelized snake case version of `name`

## Why

Error codes

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -171,6 +171,11 @@ decamelize@^1.2.0:
   resolved "https://registry.yarnpkg.com/decamelize/-/decamelize-1.2.0.tgz#f6534d15148269b20352e7bee26f501f9a191290"
   integrity sha1-9lNNFRSCabIDUue+4m9QH5oZEpA=
 
+decamelize@^4.0.0:
```
Comment (ShannonC0de) [inline yarn.lock:174]: do we expect the package.json to change as a result of this and be committed? I know this is the source of truth but don't recall whether we include the package.json which changes as a result of this in source control

**Review by ShannonC0de — APPROVED** [review]
aside from my question about package.json LGTM 🍡

File: yarn.lock (hunk 1 above)
Comment (ajpawlicki) [reply yarn.lock:174]: We are adding `decamelize` to the package.json. Is that what you're asking?

File: package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
     "mocha": "^7.1.0"
   },
   "dependencies": {
-    "@hapi/boom": "^7.4.0"
+    "@hapi/boom": "^7.4.0",
+    "decamelize": "^4.0.0"
```
Comment (ajpawlicki) [inline package.json:37]: @ShannonC0de it's being added here

File: yarn.lock (hunk 1 above)
Comment (ShannonC0de) [reply yarn.lock:174]: yes, sorry you are good!

**Comment by ajpawlicki** [comment]
This will be a major version change

**Pull request merged**